lib/bootloader_setup.pm

```diff
@@ -916,30 +916,33 @@ sub tianocore_http_boot {
 
 sub zkvm_add_disk {
     my ($svirt) = @_;
-    if (my $hdd = get_var('HDD_1')) {
-        my $basename = basename($hdd);
-        my $basedir  = svirt_host_basedir();
-        my $hdd_dir  = "$basedir/openqa/share/factory/hdd";
-        my $hdd_path = $svirt->get_cmd_output("find $hdd_dir -name $basename | head -n1 | tr -d '\n'");
-        die "Unable to find image $basename in $hdd_dir" unless $hdd_path;
-        diag("HDD path found: $hdd_path");
-        if (get_var('PATCHED_SYSTEM')) {
-            diag('in patched systems just load the patched image');
-            my $name        = $svirt->name;
-            my $patched_img = "$zkvm_img_path/$name" . "a.img";
-            $svirt->add_disk({file => $patched_img, dev_id => 'a'});
+    my $dev_id = 'a';
+    my $numdisks = get_var('NUMDISKS') || '1';
+
+    foreach my $n (1 .. $numdisks) {
+        if (my $hdd = get_var('HDD_' . $n)) {
+            my $basename = basename($hdd);
+            my $basedir  = svirt_host_basedir();
+            my $hdd_dir  = "$basedir/openqa/share/factory/hdd";
+            my $hdd_dir  = "/var/lib/libvirt/images";
```

Comment:
You're using the same variable twice, or rather you're trying to change the behavior here?

Reply:
Oops, seems that was a remainder of obsolete testcode. I've removed it now.